src/purchase-client.ts

```diff
@@ -0,0 +1,203 @@
+   * @example
+   * client.purchase("XXSKU000");
+   */
+  public purchase(identifier: string) {
+    if (!window.Twitch.ext.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+
+    window.Twitch?.ext?.bits.useBits(identifier);
+  }
+ 
+  /**
+   * Returns a list of Twitch Bits Products for the current channel.
+   * 
+   * @async
+   * @since 2.4.0
+   * 
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   *
+   * @return {Promise<[]TwitchBitsProduct>} Resolves with an array of {@link TwitchBitsProduct}
+   * objects for each available sku.
+   *
+   * @example
+   * const products = client.getProducts();
+   */
+  public getProducts(): Promise<TwitchBitsProduct[]> {
+    const target = window.Twitch?.ext?.bits;
+    if (!window.Twitch.ext.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+    
+    return target.getProducts();
+  } 
+
+  /**
+   * onUserPurchase is the interface for adding a post transaction callback.
+   *
+   * @since 2.4.0
+   *
+   * @param callback a function(body)
+   *
+   * @example
+   * client.onUserPurchase(() => {
+   *  console.log("Transaction finished!");
+   * });
+   */
+  public onUserPurchase(callback: (tx: Transaction) => void): void {
+    window.Twitch.ext.bits.onTransactionComplete(
+      (tx: TwitchBitsTransaction) => {
+        if (tx.initiator.toLowerCase() === "current_user") {
+          callback(tx);
+        }
+      }
+    );
+  }
+}
+
+// DevPurchaseClient implements the basic 'purchase client' interface.
+// This is used by SDK to provide low-level access to stubbed transactions.
+class DevPurchaseClient implements PurchaseClient {
+  /**
+   * purchase will start the Dev purchase transaction.
+   *
+   * @since 2.4.0
+   *
+   * @throws {SDKError} Will throw an error if the MuxySDK didn't load.
+   * 
+   * @param identifier the twitch sku of the item to be purchased.
+   *
+   * @example
+   * client.purchase("TESTSKU00");
+   */
+  public purchase(identifier: string) {
+    // TODO: implement some type of purchase for dev testing?
+  }
+ 
+  /**
+   * Returns a list of Twitch Bits Products for the current channel.
+   * 
+   * @async
+   * @since 2.4.0
+   *
+   * @return {Promise<[]Product>} Resolves with an array of {@link Product}
+   * objects for each available sku.
+   *
+   * @example
+   * window.MEDKIT_PURCHASE_ITEMS = [{
+   *   sku: "TESTSKU01",
+   *   displayName: "Test SKU 01",
+   *   cost: {
+   *       amount: 1,
+   *       type: "test-cost",
+   *     },
+   *   },
+   *   {
+   *     sku: "TESTSKU02",
+   *     displayName: "Test SKU 02",
+   *     cost: {
+   *      amount: 2,
+   *      type: "test-cost",
+   *   }
+   * ];
+   *
+   * const products = client.getProducts();
+   */
+  public getProducts(): Promise<Product[]> {
+    if ("MEDKIT_PURCHASABLE_ITEMS" in window) {
+      return new Promise((resolve, reject) => {
+        resolve(window.MEDKIT_PURCHASABLE_ITEMS);
+      });
```

Comment:
You can shorten this to `return Promise.resolve(window.MEDKIT_PURCHASABLE_ITEMS)`